assets/js/components/Board/index.js

```diff
@@ -1,46 +1,109 @@
+        .then((response) => {
+            let data = response.data;
+            let boardData = data.board;
+            let currentPlayer = data.current_player;
+            let gameStatus = data.game_status;
+            let reduceBoard = Object.values(boardData)
+            this.setState({
+                gameStatus: gameStatus,
+                moves: reduceBoard,
+                currentPlayer: currentPlayer,
+            })
+        }).catch(error => console.log(error))
+    }
+
+    getCurrentPlayer() {
+        if (this.state.currentPlayer == "X") {
+            this.setState({
+                currentPlayer: "O",
+                message: "Player two,"
+            })
+            return "X"
+        } 
+        else {
+            this.setState({
+                currentPlayer: "X",
+                message: "Player one,"
+            })
+            return "O"
+        }
+            
+    }
+
+    removeErrorMessage() {
+        this.setState({error: " "});
+    }
+
+    selectSquare(positionNumber) {
         const moves = this.state.moves.slice();
-        moves[position_number] = "X";
-        this.setState({
-            moves: moves
-        });
-        this.postNewMark()
+        console.log(moves)
+        if (moves[positionNumber] == null) {
+            moves[positionNumber] = this.getCurrentPlayer();
+            this.postNewMark(positionNumber);
+        }
+        else {
+            this.setState({error: "Oh no, select an empty space!"});
+            setTimeout(()=> this.removeErrorMessage(), 1000);
+        }
     }
 
-    postNewMark() {
-        axios.post('/api/boardstatus', {
+    postNewMark(requestedMove) {
+        axios.post('/api/createmove', {
             headers: {"Content-Type": "application/json"},
             data: {
-                moves: this.state.moves
+                board: this.state.moves,
+                gameStatus: this.state.gameStatus,
+                currentPlayer: this.state.currentPlayer,
+                incomingMove: requestedMove
             }
-        });
+        })
+        .then((response) => {
+            let data = response.data;
+            let boardData = data.board;
+            let gameStatus = data.game_status;
+            let reduceBoard = Object.values(boardData)
+            this.setState({
+                gameStatus: gameStatus,
+                moves: reduceBoard
+            })
+        }).catch(error => console.log(error));
     }
 
-    renderSquare(position_number) {
+    renderSquare(positionNumber) {
         
         return (
           <Square
-            onClick={() => this.selectSquare(position_number)}
-            value={this.state.moves[position_number] || "-"}
+            onClick={() => this.selectSquare(positionNumber)}
+            value={this.state.moves[positionNumber] || "-"}
           />
         );
     }
     
     render() {
         return (
             <div>
+                <section className="message-box">
```

Comment:
Should message box class be inside message?

Reply:
Moved that over to message. Thanks!